packages/typespec-client-generator-core/src/types.ts

```diff
@@ -563,18 +565,51 @@ export function getSdkModelWithDiagnostics(
 
 function getSdkEnumValueType(
   context: TCGCContext,
-  type: EnumMember | StringLiteral | NumericLiteral
+  values:
+    | IterableIterator<EnumMember>
+    | IterableIterator<UnionEnumVariant<string>>
+    | IterableIterator<UnionEnumVariant<number>>
 ): SdkBuiltInType {
   let kind: "string" | "int32" | "float32" = "string";
```

Comment:
what about int64 and float64?

Reply:
This is only for enum value. Since we could not get exact type for enum value, we just use int32 or float32 to be the enum value type. And I believe it is the most widely used case.

Reply:
Got it. emitter can enlarge the type if found it to be "fixed enum".